RecoHGCal/TICL/plugins/TICLSeedingRegionProducer.cc

```diff
@@ -56,12 +60,9 @@ TICLSeedingRegionProducer::TICLSeedingRegionProducer(const edm::ParameterSet& ps
 void TICLSeedingRegionProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
   desc.add<int>("algo_verbosity", 0);
-  desc.add<edm::InputTag>("tracks", edm::InputTag("generalTracks"));
-  desc.add<std::string>("cutTk",
-                        "1.48 < abs(eta) < 3.0 && pt > 1. && quality(\"highPurity\") && "
-                        "hitPattern().numberOfLostHits(\"MISSING_OUTER_HITS\") < 5");
-  desc.add<std::string>("propagator", "PropagatorWithMaterial");
   desc.add<int>("algoId", 1);
+  desc.add<edm::ParameterSetDescription>("seedTiclByTracksConfig", SeedingRegionByTracks::makePSetDescription());
+  desc.add<edm::ParameterSetDescription>("seedTiclByHFConfig", SeedingRegionByHF::makePSetDescription());
   descriptions.add("ticlSeedingRegionProducer", desc);
 }
 
```

Comment:
Given the amount and kind of changes, we should rather be using the standard `CMSSW` plugin mechanism, including its `fillDescription` handling, rather than the proposed solution.
See [here](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#A_Plugin_Module_Using_Another_He) for a detailed description of how to handles automatic `PSet` generation for plugins.
The same approach has been already used for the clustering phase.

Reply:
I saw you you did with the CLUEalgo; I agree would be better, let's my try to follow the wiki you pointed